src/shortcuts/__tests__/undo.ts

```diff
@@ -386,3 +386,50 @@ it('undo redo importText action', () => {
 
   expect(exportedAfterRedo).toEqual(expectedOutputAfterRedo)
 })
+
+it('cursor should restore to same thought if the thought has been edited after undo', () => {
+  const store = createTestStore()
+
+  store.dispatch([
+    newThought({ value: 'a' }),
+    newThought({ value: 'b' }),
```

Comment:
It would be better to emulate the user's action exactly by creating an empty thought, editing it to `a`, moving the cursor, creating another empty thought, and editing it to `b`. 

I abbreviate the steps to reproduce for readability, so apologies for the confusion.